R/mod_marg.R

```diff
@@ -36,19 +36,32 @@ mod_marg2 <- function(mod, var_interest,
                       at = NULL, base_rn = 1,
                       at_var_interest = NULL){
 
+  stopifnot('glm' %in% class(mod))
+
   df <- mod$data[, all.vars(mod$formula)]
   df <- df[complete.cases(df), ]
 
   stopifnot(
-    'glm' %in% class(mod),
     var_interest %in% names(df),
     all(names(at) %in% names(df))
-    # TODO: warning if at contains extrapolated values
   )
+
+  # Check for extrapolated values
+  for(i in seq_along(at)){
+    if(is.numeric(df[[names(at)[i]]]) &
+       ! all(at[[i]] <= max(df[[names(at)[i]]]) &
+             at[[i]] >= min(df[[names(at)[i]]])))
+      warning(sprintf("Not all values in 'at' are in the range of '%s'", names(at)[i]))
+  }
+
```

Comment:
Loops through the "at" list and, for numeric variables only, checks that all the values are within the range of the data.